server/scripts/setupErrorHandlers.ts

```diff
@@ -1,19 +1,53 @@
-import { factory, formatFilename } from '../../shared/logging';
-const log = factory.getLogger(formatFilename(__filename));
+import Rollbar from 'rollbar';
+import { ROLLBAR_SERVER_TOKEN } from '../config';
+import { ServerError, AppError } from '../util/errors';
 
-const setupErrorHandlers = (app, rollbar) => {
-  // catch 404 and forward to error handler
+const setupErrorHandlers = (app) => {
+  // Handle 404 errors
   app.use((req, res, next) => {
-    const err : any = new Error('Not Found');
-    err.status = 404;
-    next(err);
+    const error: any = new Error('Not found');
+    error.status = 404;
+    next(error);
   });
-  // error handler
-  if (process.env.NODE_ENV === 'production') app.use(rollbar.errorHandler());
-  app.use((err, req, res, next) => {
-    log.error(err.stack);
-    res.status(err.status || 500);
-    res.json({ error: err.message });
+
+  // Rollbar notifications
+  const rollbar = new Rollbar({
+    accessToken: ROLLBAR_SERVER_TOKEN,
+    environment: process.env.NODE_ENV,
+    captureUncaught: true,
+    captureUnhandledRejections: true,
+  });
+  app.use(rollbar.errorHandler());
+
+  // Handle server and application errors
+  app.use((error, req, res, next) => {
+    if (error instanceof ServerError) {
+      rollbar.error(error); // expected server error
+      res.status(error.status).send({
+        error: {
+          status: error.status,
+          // Use external facing error message
+          message: 'Server error, please try again later.',
+        },
+      });
+    } else if (error instanceof AppError) {
+      rollbar.log(error); // expected application/user error
+      res.status(error.status).send({
+        error: {
+          status: error.status,
+          message: error.message,
+        },
+      });
+    } else {
+      rollbar.critical(error); // unexpected error
+      res.status(error.status || 400);
```

Comment:
We should probably default to 500 for server errors + also return a hard-coded message, as critical errors may contain stack traces.

Reply:
That makes sense, will make those changes. As we get these errors, we'll have to sort out the actual server ones from normal application ones.